pkg/decoder/line_buffer.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package decoder
+
+import (
+	"bytes"
+)
+
+// LineBuffer accumulates lines in buffer escaping all '\n'
+// and accumulates the total number of bytes of all lines in line representation (line + '\n') in contentLen
+// to form and forward outputs to outputChan
+type LineBuffer struct {
+	outputChan chan *Output
+	buffer     *bytes.Buffer
+	contentLen int
+}
+
+// NewLineBuffer returns a new LineBuffer
+func NewLineBuffer(outputChan chan *Output) *LineBuffer {
+	buffer := bytes.Buffer{}
+	return &LineBuffer{
+		outputChan: outputChan,
+		buffer:     &buffer,
+	}
+}
+
+// IsEmpty returns true if buffer does not contain any line
+func (l *LineBuffer) IsEmpty() bool {
+	return l.contentLen == 0
+}
+
+// Length returns the length of buffer which might be different from contentLen because of escaped '\n'
+func (l *LineBuffer) Length() int {
+	return l.buffer.Len()
+}
+
+// Add stores line in buffer
+func (l *LineBuffer) Add(line *Line) {
+	l.buffer.Write(line.content)
+	l.contentLen += len(line.content) + 1 // add 1 for '\n'
+}
+
+// AddEndOfLine stores an escaped '\n' in buffer
+func (l *LineBuffer) AddEndOfLine() {
+	l.buffer.Write([]byte(`\n`))
+}
+
+// AddPieceOfLine stores line in buff
+func (l *LineBuffer) AddPieceOfLine(line *Line) {
```

Comment:
Why do we call it `AddPieceOfLine` vs `AddLine`?

Reply:
Got it, how about `AddIncompleteLine`?

Reply:
I agree, I could be more specific here, it's just a piece because it did not end with '\n', since it was too long, it has been sent to `MultiLineHandler`

Reply:
ok, let's either add a small comment or find a better name. This is the right behavior, but a bit confusing